File: test1.c
#include <mpi.h>
#include <stdio.h>

int main(int argc, char **argv)
{
    int procNum, procRank, tmp;
    MPI_Status status;
    MPI_Init(&argc, &argv);
    MPI_Comm_size(MPI_COMM_WORLD, &procNum);
    MPI_Comm_rank(MPI_COMM_WORLD, &procRank);
    printf("Hello from process %i \n", procRank);
    MPI_Finalize();
    return 0;
}

File: L8/test1.f
      program main_mpi
        implicit none
        include 'mpif.h'
       
        integer myid, nproc, ierr

        call MPI_Init(ierr)
        call MPI_Comm_rank(MPI_COMM_WORLD, myid, ierr)
        call MPI_Comm_size(MPI_COMM_WORLD, nproc, ierr)
        print'(a8,i1,a4,i1)', "process ", myid, " of ", nproc
        call MPI_Finalize(ierr)
      end program
